### PR TITLE
Site tool polishing

### PR DESCRIPTION
**Pull request by jacksonsj**

### CODE REVIEW

File: R/utilities.R [hunk 1]
```diff
@@ -253,5 +259,137 @@ exportPackr <- function(data, output_path, type, datapack_name) {
...
+  
+  # Add new new countries as PSNUs
+    newCountries <- datapackr::configFile %>%
+      dplyr::filter(
+        model_uid == datapack_uid,
+        Currently_in_DATIM == "N",
+        isMil == "0") %>%
+      dplyr::select(countryName, countryUID)
+    
+  if (NROW(newCountries) != 0) {
```
Comment (jason-p-pickering) [inline R/utilities.R]: Could `newCountries` be NULL or NA?

File: R/utilities.R [hunk 2]
```diff
@@ -253,5 +259,137 @@ exportPackr <- function(data, output_path, type, datapack_name) {
...
+                              uid = milPSNUuid,
+                              name = milPSNU))
+  
+  # Patch for Suriname and Jamaica
+  if(datapack_uid == "Caribbean_Data_Pack") {
+    patchList <- c("Suriname","Jamaica")
+    
+    JamaicaSuriname = NULL
+    
+    for (i in 1:length(patchList)) {
```
Comment (jason-p-pickering) [inline R/utilities.R]: Replace with `seq_along`

File: R/mapMechanisms.R [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+    
+  # Add Home tab ####
+    wb <- datapackr::writeHomeTab(wb = wb,
+                                  data_pack_uid = datapack_uid,
+                                  type = "Mechanism Map")
+    
+  # Add other tabs ####
+    sheet_names <- c("Old Mechs", "New Mechs", "Validations", "Mechanism Map")
+    visible <- c(TRUE, TRUE, FALSE, TRUE)
+    for (i in 1:length(sheet_names)) {
```
[the remark this answers is not in the document]
Comment (jacksonsj) [reply R/mapMechanisms.R]: Done

File: R/packFrame.R [hunk 4]
```diff
@@ -145,8 +145,8 @@ frameDataSheet <- function(wb, sheet, type = "Data Pack") {
 #' @param wb An Openxlsx workbook object
 #'
 addValidationsSite <- function(wb) {
-  openxlsx::addWorksheet(wb, sheetName = "Validations")
-  openxlsx::sheetVisibility(wb)[which(names(wb) == "Validations")] <- "veryHidden"
+  openxlsx::addWorksheet(wb, sheetName = "Validations",visible = FALSE)
+  #openxlsx::sheetVisibility(wb)[which(names(wb) == "Validations")] <- "veryHidden"
```
[the remark this answers is not in the document]
Comment (jacksonsj) [reply R/packFrame.R]: Done

File: R/packFrame.R [hunk 5]
```diff
@@ -276,8 +276,8 @@ packFrame <- function(datapack_uid, type = "Data Pack") {
       openxlsx::addWorksheet(wb, sheetName = "Site List")
     
     # Add Mech List tab
-      openxlsx::addWorksheet(wb, sheetName = "Mechs")
-      openxlsx::sheetVisibility(wb)[which(names(wb) == "Mechs")] <- "veryHidden"
+      openxlsx::addWorksheet(wb, sheetName = "Mechs", visible = FALSE)
+      #openxlsx::sheetVisibility(wb)[which(names(wb) == "Mechs")] <- "veryHidden"
```
[the remark this answers is not in the document]
Comment (jacksonsj) [reply R/packFrame.R]: Done

File: R/packSiteTool.R [hunk 6]
```diff
@@ -54,7 +54,7 @@ colorCodeSites <- function(wb, sheet, cols, rows) {
 #' @param d A datapackr list object.
 #' 
 write_site_level_sheet <- function(wb, sheet, d) {
-print(sheet)
+  print(sheet)
```
[the remark this answers is not in the document]
Comment (jacksonsj) [reply R/packSiteTool.R]: Done - removed in all places

File: R/packSiteTool.R [hunk 7]
```diff
@@ -288,8 +299,10 @@ packSiteTool <- function(d) {
       dplyr::filter(data_pack_name == d$info$datapack_name) %>%
       dplyr::pull(country_uid)
     
-    siteList <- datapackr::getSiteList(country_uids,
-                                       include_mil = TRUE) %>%
+    sites <- datapackr::getSiteList(country_uids,
```
[the remark this answers is not in the document]
Comment (jacksonsj) [reply R/packSiteTool.R]: Done in all code for all internal package functions

File: R/parseDataPack.R [hunk 8]
```diff
@@ -779,25 +783,88 @@ packSUBNAT_IMPATT <- function(d) {
     dplyr::mutate(
       value = round_trunc(value),
       period = "2019Oct",
-      attributeoptioncombo = "HllvX50cXC0"
+      attributeOptionCombo = "HllvX50cXC0"
```
[the remark this answers is not in the document]
Comment (jacksonsj) [reply R/parseDataPack.R]: Like this?: 99522f18bbd29b3c300950dd78bde5fe536dcebb

File: R/parseDataPack.R [hunk 9]
```diff
@@ -779,25 +783,88 @@ packSUBNAT_IMPATT <- function(d) {
...
+  }
+  
+  if (type == "SUBNAT_IMPATT") {
+    importFile <- packSUBNAT_IMPATT(data)
+  }
+  
+  if (type == "PSNUxIM") {
+    importFile <- data %>%
+      dplyr::mutate(
+        period = "2019Oct",
```
[the remark this answers is not in the document]
Comment (jacksonsj) [reply R/parseDataPack.R]: Better to set this as a package setting à la [datimvalidation::api_version()](https://github.com/jason-p-pickering/datim-validation/blob/c62eba5621fb786d03fe0bd691a77cfa80866f7b/R/internals.R#L101)? Or global variable? I see what you're saying about making this more plug and play for future FYs.

File: R/utilities.R [hunk 10]
```diff
@@ -142,20 +144,24 @@ getMilitaryNodes <- function() {
...
-      dots <-
-        stats::setNames(list(lazyeval::interp(
-          ~ magrittr::use_series(from, x), x = as.name(col)
-        )), col)
-      to <- to %>%
-        dplyr::mutate_(.dots = dots)
-    } else {
-      next
+  
+  if (length(cols) != 0) {
```
[the remark this answers is not in the document]
Comment (jacksonsj) [reply R/utilities.R]: See 01f3485482516937334942871fd623ffd8a10125. Better?

File: R/utilities.R [hunk 11]
```diff
@@ -219,14 +225,14 @@ exportPackr <- function(data, output_path, type, datapack_name) {
...
-  if (type %in% c("Site Tool", "Data Pack")) {
+  if (type %in% c("Site Tool", "Data Pack", "Mechanism Map")) {
     if (class(data) != "Workbook") {
       stop("Output type and data do not match!")
     }
     
     output_file_name <- packName(output_path, type, datapack_name, extension = ".xlsx")
     
-    openxlsx::saveWorkbook(wb, output_file_name, overwrite = TRUE)
+    openxlsx::saveWorkbook(data, output_file_name, overwrite = TRUE)
```
[the remark this answers is not in the document]
Comment (jacksonsj) [reply R/utilities.R]: Clearer? 4cf7f715bab4b0bd1c65afe086b96686b76e9096

File: R/utilities.R [hunk 12]
```diff
@@ -253,5 +259,137 @@ exportPackr <- function(data, output_path, type, datapack_name) {
...
+#' additional PSNUs not currently in DATIM as needed.
+#' 
+#' @param datapack_uid A unique ID specifying the PEPFAR Operating Unit 
+#' the Site Tool belongs to.
+#' 
+#' @return Data frame of PSNUs
+#' 
+getPSNUs <- function(datapack_uid) {
+  # Pull PSNUs from DATIM SQL view
+    DATIMcountryNameString <- datapackr::configFile %>%
```
[the remark this answers is not in the document]
Comment (jacksonsj) [reply R/utilities.R]: 865beaef2399605ee207459317bbfbec56073898

File: R/utilities.R [hunk 13]
```diff
@@ -253,5 +259,137 @@ exportPackr <- function(data, output_path, type, datapack_name) {
...
+      dplyr::filter(model_uid == datapack_uid,
+             Currently_in_DATIM == "Y") %>%
+      dplyr::select(countryName) %>%
+      dplyr::mutate(
+        countryName = stringr::str_replace(countryName,
+                                           "d'Ivoire",
+                                           "d Ivoire")) %>%
+      unique() %>%
```
[the remark this answers is not in the document]
Comment (jacksonsj) [reply R/utilities.R]: 1c0fb0a5c07f06699121bd908070c35d74b5d66a

File: R/utilities.R [hunk 14]
```diff
@@ -253,5 +259,137 @@ exportPackr <- function(data, output_path, type, datapack_name) {
...
+      dplyr::bind_rows(milPSNUs %>%
+                       dplyr::select(country = DataPack_name,
+                              uid = milPSNUuid,
+                              name = milPSNU))
+  
+  # Patch for Suriname and Jamaica
+  if(datapack_uid == "Caribbean_Data_Pack") {
+    patchList <- c("Suriname","Jamaica")
+    
+    JamaicaSuriname = NULL
```
[the remark this answers is not in the document]
Comment (jacksonsj) [reply R/utilities.R]: Let's discuss this one in the context of https://github.com/pepfar-datim/Global/issues/4655

File: data-raw/run.R [hunk 15]
```diff
@@ -1,6 +1,22 @@
-
+support_files <- "/Users/scott/Google Drive/PEPFAR/COP Targets/COP 19/3) Testing & Deployment/Support Files"
```
[the remark this answers is not in the document]
Comment (jacksonsj) [reply data-raw/run.R]: Just an example execution file

File: data-raw/runMechMap.R [hunk 16]
```diff
@@ -0,0 +1,16 @@
+output_path <- "/Users/scott/Google Drive/PEPFAR/COP Targets/COP 19/5) Maintenance & Support/Mechanism Maps"
```
[the remark this answers is not in the document]
Comment (jacksonsj) [reply data-raw/runMechMap.R]: Just an example execution file

File: R/mapMechanisms.R [hunk 17]
```diff
@@ -0,0 +1,273 @@
...
+      openxlsx::dataValidation(
+        wb, sheet = "Mechanism Map",
+        cols = 3, rows = 4:200,
+        type = "list",
+        value = 'INDIRECT("indicators[indicator]")'
+      )
+    ## Types
+      openxlsx::dataValidation(
+        wb, sheet = "Mechanism Map",
+        cols = 4, rows = 4:200,
```
[the remark this answers is not in the document]
Comment (jacksonsj) [reply R/mapMechanisms.R]: You may have to explain your question more to me.

File: R/utilities.R [hunk 18]
```diff
@@ -253,5 +259,137 @@ exportPackr <- function(data, output_path, type, datapack_name) {
...
+             Currently_in_DATIM == "Y") %>%
+      dplyr::select(countryName) %>%
+      dplyr::mutate(
+        countryName = stringr::str_replace(countryName,
+                                           "d'Ivoire",
+                                           "d Ivoire")) %>%
+      unique() %>%
+      dplyr::pull(countryName) %>%
+      paste(collapse = ",") %>%
+      stringr::str_replace_all("&","%26") %>%
```
[the remark this answers is not in the document]
Comment (jacksonsj) [reply R/utilities.R]: ed5bc831dc1b3551646e948c6af01190ed5ecf4e

File: R/utilities.R [hunk 19]
```diff
@@ -253,5 +259,137 @@ exportPackr <- function(data, output_path, type, datapack_name) {
...
+                    "/sqlViews/PjjAyeXUbBd/data.json?",
+                    "fields=operating_unit,uid,name",
+                    "&filter=name:!like:_Military",
+                    "&filter=operating_unit:in:[",
+                    DATIMcountryNameString,"]") %>%
+      URLencode() %>%
+      httr::GET() %>%
+      httr::content(., "text") %>%
+      jsonlite::fromJSON(., flatten = TRUE)
+    PSNUs <- as.data.frame(PSNUs$rows,stringsAsFactors = FALSE) %>%
```
[the remark this answers is not in the document]
Comment (jacksonsj) [reply R/utilities.R]: Struggling with what to do immediately after this for setting column names

File: R/getMechList.R [hunk 20]
```diff
@@ -11,21 +11,24 @@
...
 #' 
 #' @return A dataframe of mechanisms, including start and end dates, mechanism
 #' code, partner name, funding agency, and related country.
 #'
 getMechList <- function(country_uids = NA,
+                        include_dedupe = FALSE,
                         FY = NA) {
   
-  datapackr::loginToDATIM(getOption("secrets"))
+  loginToDATIM(getOption("secrets"))
```
Comment (jason-p-pickering) [inline R/getMechList.R]: Remove this. Should be part of the automation code. Ideally, on line 42, you should check and be sure the result of the API call gives you an HTTP status of 200.

File: R/getMechList.R [hunk 21]
```diff
@@ -35,7 +38,7 @@ getMechList <- function(country_uids = NA,
                !all(is.na(country_uids)),
                paste0("&filter=categoryOptions.organisationUnits.id:in:[",
                       paste0(country_uids,collapse = ","),"]"),
-                      ""),
+               "&filter=code:!in:[00000,00001,00100,00200]"),
```
Comment (jason-p-pickering) [inline R/getMechList.R]: Maybe just DO NOT get the MOH mechanisms? It seems that dedupe would be valid mechanisms actually to me?

**Pull request merged**